src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php

```diff
@@ -251,9 +252,24 @@ private function guessHandledClasses(\ReflectionClass $handlerClass, string $ser
     private function registerReceivers(ContainerBuilder $container, array $busIds)
     {
         $receiverMapping = [];
+        $failureTransportsMap = [];
+
+        $globalFailureTransportId = 'messenger.failure_transports.default';
+        if ($container->hasAlias($globalFailureTransportId)) {
+            $globalFailureTransport = (string) $container->getAlias($globalFailureTransportId);
+            if (null !== $globalFailureTransport) {
+                $failureTransportsMap[$globalFailureTransport] = new Reference('messenger.transport.'.$globalFailureTransport);
+            }
+        }
```

Comment:
I'm not sure what this section is going. Is it... setting the "failure transport" of the "global failure transport" to itself? If so, why?